js/classes/asset_manager.js

```diff
@@ -121,27 +123,56 @@ class AssetManager {
     };
 
     /**
-     * This method plays the audio associated with the given path.
+     * Plays the audio associated with the given path.
      * @param {string} path The filepath of the audio you are trying to play.
      * @param {number} volume The volume to which you want to set the audio.
-     * @param {boolean} loop True if you want the audio to loop, false otherwise.
      */
-    playAudio(path, volume, loop) {
-        loop ?? (loop = false);
+    playSFX(path, volume) {
         const audio = this.cache[path];
         audio.currentTime = 0;
         audio.volume = volume;
 
         audio.play();
-        if (loop) {
-            audio.addEventListener("ended", () => {
-                audio.play();
-            });
-        }
     };
 
     /**
-     * This method stops the audio associated with the given path.
+     * Plays the audio associated with the given path.
+     * @param {string} path The filepath of the audio you are trying to play.
+     * @param {number} volume The volume to which you want to set the audio.
+     */
+    playMusic(path, volume) {
+        const audio = this.cache[path];
+        audio.currentTime = 0;
+        audio.volume = volume;
+        this.currentMusic = audio;
+
+        audio.play();
+        audio.addEventListener("ended", () => { //! might have a bug where pauseMusic "ends" the music. Check it out
+            audio.play();
+        });
+    };
+
+    /**
+     * Pauses the currently playing background music.
+     */
+    pauseMusic() {
+        if (this.currentMusic) {
```

Comment:
Would it make more sense to use a try catch here? Not sure it matters either way.